lib/pbench/server/api/resources/elasticsearch_api.py

```diff
@@ -0,0 +1,82 @@
+import requests
+from flask_restful import Resource, abort
+from flask import request, make_response
+from flask_cors import cross_origin
+
+
+class Elasticsearch(Resource):
+    """Elasticsearch API for post request via server."""
+
+    def __init__(self, config, logger):
+        self.logger = logger
+        self.es_host = config.get_conf(__name__, "elasticsearch", "host", self.logger)
+        self.es_port = config.get_conf(__name__, "elasticsearch", "port", self.logger)
+
+    @cross_origin()
+    def post(self):
+
+        self.elasticsearch = f"http://{self.es_host}:{self.es_port}"
+
+        json_data = request.get_json(silent=True)
+        if not json_data:
+            message = "Invalid json object in the query"
+            self.logger.warning(f"{message}: {request.url}")
+            abort(400, message=message)
+
+        if not json_data["indices"]:
+            message = "Missing indices path in the post request"
+            self.logger.warning(f"{message}")
+            abort(400, message=f"{message}")
+
+        try:
+            # query Elasticsearch
+            if "params" in json_data:
+                url = (
+                    f"{self.elasticsearch}/{json_data['indices']}?{json_data['params']}"
+                )
+            else:
+                url = f"{self.elasticsearch}/{json_data['indices']}"
+
+            if "payload" in json_data:
+                es_response = requests.post(url, json=json_data["payload"])
+            else:
+                self.logger.debug(
+                    "No payload found in Elasticsearch post request json data"
+                )
+                es_response = requests.get(url)
```

Comment:
You need to add the `es_response.raise_for_status()` here to activate the exception transformation. That's in my PR; however since you've moved everything your merge is going to be messy and it could easily be overlooked!

Reply:
Thanks for the heads up